### PR TITLE
feat: enable multi-type toolbar items

### PR DESCRIPTION
**Pull request by SamCreasey**
Remove `toolbarFlyoutItems` prop and extend the `toolbarItems` prop of the BlockItemWrapper to enable users more flexibility with toolbar items while keeping the strict typing of the previous implementation.
Users can now an array of objects, each of which should have a `type` property that is one of "dragHandle", "menu", "flyout", "button". These button object can be in any order, meaning that any item is able to open it's own flyout, be a dragHandle, open a menu or be a simple button

### CODE REVIEW

File: .changeset/olive-roses-mix.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
```
Comment (ragi96) [inline .changeset/olive-roses-mix.md:1]: Probably we should add an example of each type

File: packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.spec.tsx [hunk 2]
```diff
@@ -33,15 +34,89 @@ describe('Toolbar', () => {
...
     });
 
     it('should throw error if toolbar does have attachments enabled without provider', () => {
-        expect(() =>
-            render(<Toolbar items={[]} flyoutMenu={{ items: [] }} attachments={{ isEnabled: true }} />),
-        ).toThrowError();
+        expect(() => render(<Toolbar items={[]} attachments={{ isEnabled: true }} />)).toThrowError();
+    });
+
+    it('should render every item type', async () => {
```
Comment (ragi96) [inline packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.spec.tsx:44]: Can we have this also tested with Keyboard? as now its only tested on click?

File: .changeset/olive-roses-mix.md (hunk 1 above)
Comment (SamCreasey) [reply .changeset/olive-roses-mix.md:1]: fixed

File: packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.spec.tsx (hunk 2 above)
Comment (SamCreasey) [reply packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.spec.tsx:45]: fixed

Comment (ragi96) [reply packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.spec.tsx:45]: Would split the ones for keyboard to an own test

Comment (SamCreasey) [reply packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar/Toolbar.spec.tsx]: fixed

**Review by ragi96 — APPROVED** [review]
LGTM

**Pull request merged**